### PR TITLE
Fix deprecation warnings (except for NSCopyBits).

### PR DESCRIPTION
**Pull request by jpetrie**
These changes fix the last remaining deprecation warnings in MacVim itself (except for the one related to `NSCopyBits`, which as documented in #88 has me stumped).

They don't address the warnings in PSMTabBarFramework, as it's a third-party library and updating it to fix the warnings might create difficulties building for older OS X SDKs.

### CODE REVIEW

File: src/MacVim/MMAppController.m [hunk 1]
```diff
@@ -1721,12 +1721,17 @@ - (void)handleGetURLEvent:(NSAppleEventDescriptor *)event
         while ((param = [enumerator nextObject])) {
             NSArray *arr = [param componentsSeparatedByString:@"="];
             if ([arr count] == 2) {
+#if MAC_OS_X_VERSION_MIN_REQUIRED >= MAC_OS_X_VERSION_10_11
+              [dict setValue:[[arr lastObject] stringByRemovingPercentEncoding]
```
Comment (splhack) [inline src/MacVim/MMAppController.m]: could you fit the indent?

**Comment by splhack** [comment]
other than that, it looks good to me

File: src/MacVim/MMAppController.m (hunk 1 above)
Comment (jpetrie) [inline src/MacVim/MMAppController.m]: Oops. Yes, I'll take care of that once I'm back at my desk.

**Comment by splhack** [comment]
@jpetrie Nice!

**Pull request merged**